packages/xpath/test/helpers.ts

```diff
@@ -194,9 +194,10 @@ export class TestContext<XForms extends boolean = false> {
 		const { message } = options;
 
 		if (Number.isNaN(expected)) {
-			expect(numberValue, message).to.be.NaN;
+			expect(numberValue, message).toBeNaN();
 		} else {
-			expect(numberValue, message).to.equal(expected);
+			// numberValue can be -0; jest's toEqual fails when -0 is compared with 0
+			expect(expected === numberValue, message).toEqual(true);
```

Comment:
💭 maybe we should change evaluator to never return -0

Reply:
No. The [spec](https://www.w3.org/TR/1999/REC-xpath-19991116/#numbers) is clear that negative zero is a distinct value, per IEEE 754 (emphasis added):

> A number represents a floating-point number. A number can have any double-precision 64-bit format IEEE 754 value [[IEEE 754]](https://www.w3.org/TR/1999/REC-xpath-19991116/#IEEE754). These include a special "Not-a-Number" (NaN) value, positive and negative infinity, and positive **and negative zero**. See [Section 4.2.3](http://java.sun.com/docs/books/jls/html/4.doc.html#9208) of [[JLS]](https://www.w3.org/TR/1999/REC-xpath-19991116/#JLS) for a summary of the key rules of the IEEE 754 standard.

I see three failures caused by the stricter check. Of those:

- One failure in `round.spec.ts` exercises an ODK extension of the standard `round`, and the behavior is consistent with the underlying [single-arity spec](https://www.w3.org/TR/1999/REC-xpath-19991116/#function-round).

- Two failures in `number-operator.spec.ts` exercise the `mod` and `*` operators. The XPath spec itself doesn't go into detail of the behavior, but the same `-0` values are exactly what the native XPath evaluator produce.

In all of these cases, I believe the functionality is correct and expected, and that the tests were previously too lax.

I think the right thing to do is to correct the expected values for those three failures, and change this to the stricter/more correct:

```suggestion
			expect(numberValue, message).toBe(expected);
```

... and probably remove the comment above it.